File: iteration2/src/MathHelper.java
package iteration2.src;

import java.util.Random;

public class MathHelper {
    private static Random rng = new Random(System.currentTimeMillis());

    private MathHelper() { }

    public static float RandomFloat(){
        return rng.nextFloat();
    }

    public static float randomFloatBetween(float min, float max){
        if(max == min)
            return max;
        if(max < min)
            throw new IllegalArgumentException("Max must be greater than min");

        return rng.nextFloat() * (max - min) + min;
    }

    public static int randomInteger(){
        return rng.nextInt();
    }

    //Upperbound is exclusive
    public static int randomIntegerBetween(int min, int max){
        if(max == min)
            return max;
        if(max < min)
            throw new IllegalArgumentException("Max must be greater than min");

        return rng.nextInt(max - min) + min;
    }

    public static float roundFloat(float num, int noOfDecimalDigits){
        if(noOfDecimalDigits > 6)
            noOfDecimalDigits = 6;

        int powOf10 = (int)Math.pow(10,noOfDecimalDigits);
        float temp = num * powOf10;

        return Math.round(temp) / (float)powOf10;
    }
}

File: iteration3/MathHelper.py
import random
import time

class MathHelper:
    def __init__(self):
        self.rng = random.Random(time.time()*1000)

    @classmethod
    def random_float(cls):
        return cls.rng.random()

    @classmethod
    def random_float_between(cls, min, max):
        if max == min:
            return max
        if max < min:
            raise ValueError("Max must be greater than min")

        return cls.rng.uniform(min, max)

    @classmethod
    def random_integer(cls):
        return cls.rng.randint(0, 2**32 - 1)

    @classmethod
    def random_integer_between(cls, min, max):
        if max == min:
            return max
        if max < min:
            raise ValueError("Max must be greater than min")

        return cls.rng.randint(min, max)

    @classmethod
    def round_float(cls, num, no_of_decimal_digits):
        if no_of_decimal_digits > 6:
            no_of_decimal_digits = 6

        pow_of_10 = 10**no_of_decimal_digits
        temp = num * pow_of_10

        return round(temp) / pow_of_10
